app/ydoc-shared/src/ast/parse.ts

```diff
@@ -613,21 +623,30 @@ export function parse(code: string, module?: MutableModule): Owned {
   return asOwned(soleStatement)
 }
 
+export interface ParseOptions {
+  /**
+   * Specifies whether the input should be parsed as if it contains the top-level statements of a file, or statements in
+   * a body block. If `undefined` or not specified, the default is `'module'`.
+   */
+  context?: 'module' | 'block' | undefined
+  module?: MutableModule | undefined
+}
+
 /** Parse a block, and return it along with a mapping from source locations to parsed objects. */
 export function parseBlockWithSpans(
   code: string,
-  inModule?: MutableModule,
+  options: ParseOptions = {},
```

Comment:
I think it may be a bit confusing. Sometimes `Block` may mean something being module or other "body block", (like here), but sometimes it may be only the "body block" (like in parseBlock).

The fact that the "module" context in ParseOptions is optional make things even worse: on the call site, we have `parseBlock` which parses code as a block, and `parseBlockWithSpans` which, without additional settings, parses as a module.

Perhaps we should have a distinction between "block" and "body block" the latter being more specific. Or use other appropriate terms.

Also, I would recommend making `context` a mandatory parameter, as I think everyone should consider if they parse module or other block: the fact that there is an actuall difference between two may be easily missed or forgotten.

Reply:
We never actually need body blocks with spans; this was just used in the implementation of `parseBlock`. Switched `parseBlock` to an independent implementation of this logic; renamed this to `parseModuleWithSpans`; removed `context` argument.